### PR TITLE
Additional tooling

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA - LW-11589 & LW-10707 & LW-11584
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/7069/11228060126/index.html) for [85da7c51](https://github.com/input-output-hk/lace/pull/1459/commits/85da7c51d91209315e25ef6067d5d930803c98f9)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 0      | 4       | 0     | 36    | ✅     |

File: packages/nami/src/ui/UpgradeToLaceHeader.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const UpgradeToLaceHeader = ({
   const location = useLocation();
 
   if (location.pathname.startsWith('/hwTab')) return null;
+  if (location.pathname.startsWith('/dapp')) return null;
```
Comment (vetalcore) [inline packages/nami/src/ui/UpgradeToLaceHeader.tsx:17]: just a suggestion
` if (location.pathname.startsWith('/hwTab') || location.pathname.startsWith('/dapp')) return null;`

File: packages/nami/src/ui/app/pages/send.tsx [hunk 2]
```diff
@@ -709,6 +708,21 @@ const Send = ({
...
+              tx.witness
+                ? (tx.witness as Cardano.Witness)
+                : { signatures: new Map() },
+            ),
+            tx.auxiliaryData
+              ? Serialization.AuxiliaryData.fromCore(tx.auxiliaryData)
+              : undefined,
+          );
+
+          return transaction.toCbor();
```
Comment (vetalcore) [inline packages/nami/src/ui/app/pages/send.tsx]: do you think it worth moving it into the adapters folder?

File: packages/nami/src/ui/app/components/transactionBuilder.tsx [hunk 3]
```diff
@@ -456,6 +488,32 @@ const TransactionBuilder = React.forwardRef<unknown, undefined>(
               });
             }
           }}
+          getCbor={async ()=> {
```
Comment (vetalcore) [inline packages/nami/src/ui/app/components/transactionBuilder.tsx:491]: would it worth trying to unify these getCbor implementations?

File: packages/nami/src/ui/app/hw/trezorTx.tsx [hunk 4]
```diff
@@ -1,36 +1,94 @@
...
+              inMemoryWallet.utxo.available$.pipe(
+                map(utxos =>
+                  utxos.find(
+                    o =>
+                      o[0].txId === txId &&
+                      o[1].value.coins === BigInt('5000000'),
+                  ),
+                ),
+                filter(isNotNil),
+                take(1),
```
Comment (vetalcore) [inline packages/nami/src/ui/app/hw/trezorTx.tsx:59]: would it make sense to extract it into the util?

File: apps/browser-extension-wallet/src/lib/scripts/background/injectUtil.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ *   - as a side effect, get feature flags from service worker by using the
+ *     specified runtime for messaging and store them in local storage
+ */
+export const getMode = async (runtime: MinimalRuntime) =>
+  //   const storedMode = localStorage.getItem(MODE_STORAGE_KEY);
+  //   if (storedMode) {
+  //     // update feature flags in local storage without blocking the return
+  //     void loadAndStoreFeatureMode(runtime);
+  //     return JSON.parse(storedMode) as WalletMode;
+  //   }
```
[the remark this answers is not in the document]
Comment (mirceahasegan) [reply apps/browser-extension-wallet/src/lib/scripts/background/injectUtil.ts]: No. Removed it: 9334cd072983bcef9e26cc3349d9daba0b03efbd

File: packages/nami/src/ui/app/pages/dapp-connector/signTx.tsx [hunk 6]
```diff
@@ -507,21 +507,12 @@ export const SignTx = ({
...
-            //   request.data.tx,
-            //   keyHashes.key,
-            //   account,
-            //   hw,
-            //   request.data.partialSign,
-            // );
-          }
-          return await request?.sign(password);
+        sign={async password => {
+          return request?.sign(password ?? '');
```
[the remark this answers is not in the document]
Comment (mirceahasegan) [reply packages/nami/src/ui/app/pages/dapp-connector/signTx.tsx]: Done here: 41318c360a665c14ba0873bd9fb602eedf81713e

File: packages/nami/src/ui/app/pages/send.tsx [hunk 7]
```diff
@@ -700,22 +702,12 @@ const Send = ({
...
-              password,
-              withSignTxConfirmation,
-              inMemoryWallet,
-            });
+          return await signAndSubmit({
+            tx,
+            password,
+            withSignTxConfirmation,
+            inMemoryWallet,
+          });
```
[the remark this answers is not in the document]
Comment (mirceahasegan) [reply packages/nami/src/ui/app/pages/send.tsx]: Done here: 41318c360a665c14ba0873bd9fb602eedf81713e

File: packages/nami/src/ui/app/pages/send.tsx (hunk 2 above)
Comment (mirceahasegan) [reply packages/nami/src/ui/app/pages/send.tsx]: I don't recognize this code. Is the question for @AngelCastilloB ?

File: packages/nami/src/ui/app/components/transactionBuilder.tsx (hunk 3 above)
Comment (mirceahasegan) [reply packages/nami/src/ui/app/components/transactionBuilder.tsx:491]: I don't recognize this code. Is the question for @AngelCastilloB ?

File: packages/nami/src/ui/app/hw/trezorTx.tsx (hunk 4 above)
Comment (mirceahasegan) [reply packages/nami/src/ui/app/hw/trezorTx.tsx:59]: I don't recognize this code. Is the question for @AngelCastilloB ?

File: packages/nami/src/ui/indexMain.tsx [hunk 8]
```diff
@@ -147,27 +142,6 @@ export const Main = () => {
             <Route exact path="/hwTab/success">
               <SuccessAndClose />
             </Route>
-            <Route exact path="/hwTab/trezorTx/:cbor/:setCollateral?">
```
Comment (AngelCastilloB) [inline packages/nami/src/ui/indexMain.tsx]: All these routes got removed, is this change correct?, this route:

> <Route exact path="/hwTab/trezorTx/:cbor/:setCollateral?">

Is needed

File: packages/nami/src/ui/app/pages/send.tsx (hunk 2 above)
Comment (AngelCastilloB) [reply packages/nami/src/ui/app/pages/send.tsx]: done here https://github.com/input-output-hk/lace/pull/1459/commits/ceaee6ba2346e1492f67ca8b9e106f2e67702558

File: packages/nami/src/ui/app/components/transactionBuilder.tsx (hunk 3 above)
Comment (AngelCastilloB) [reply packages/nami/src/ui/app/components/transactionBuilder.tsx]: done here https://github.com/input-output-hk/lace/pull/1459/commits/ceaee6ba2346e1492f67ca8b9e106f2e67702558

File: packages/nami/src/ui/app/hw/trezorTx.tsx (hunk 4 above)
Comment (AngelCastilloB) [reply packages/nami/src/ui/app/hw/trezorTx.tsx]: done here https://github.com/input-output-hk/lace/pull/1459/commits/ceaee6ba2346e1492f67ca8b9e106f2e67702558

File: packages/nami/src/ui/indexMain.tsx (hunk 8 above)
Comment (AngelCastilloB) [reply packages/nami/src/ui/indexMain.tsx]: I added the route back here https://github.com/input-output-hk/lace/pull/1459/commits/ceaee6ba2346e1492f67ca8b9e106f2e67702558 , but we should check that all the other routes that got removed are no longer needed

File: packages/nami/src/features/ada-handle/config.ts [hunk 9]
```diff
@@ -3,7 +3,10 @@ import { Wallet } from '@lace/cardano';
 
 export const ADA_HANDLE_POLICY_ID = Wallet.ADA_HANDLE_POLICY_ID;
 export const isAdaHandleEnabled = process.env.USE_ADA_HANDLE === 'true';
-export const HANDLE_SERVER_URLS: Record<Cardano.NetworkMagics, string> = {
+export const HANDLE_SERVER_URLS: Record<
+  Cardano.NetworkMagics,
+  string | undefined
```
Comment (mirceahasegan) [inline packages/nami/src/features/ada-handle/config.ts]: Why allow `undefined`? Later it's used with non-null assertion.

File: packages/nami/src/ui/app/components/asset.tsx [hunk 10]
```diff
@@ -187,7 +188,7 @@ const Asset = ({
               size="xs"
               rightIcon={<BsArrowUpRight />}
               onClick={e => {
-                setValue({ ...value, assets: [asset] });
+                setValue({ ...value, assets: [asset as AssetInput] });
```
Comment (mirceahasegan) [inline packages/nami/src/ui/app/components/asset.tsx:191]: This looks suspicious. The props type was changed to `NamiAsset`, but here it's casted to AssetInput.

File: packages/nami/src/ui/app/components/assetsViewer.tsx [hunk 11]
```diff
@@ -129,12 +130,16 @@ const AssetsGrid = ({ assets }) => {
...
-              mt={index > 0 && 4}
+              mt={(index > 0 && 4) || undefined}
               display="flex"
               alignItems="center"
               justifyContent="center"
             >
-              <Asset cardanoCoin={cardanoCoin} asset={asset} enableSend />
+              <Asset
+                cardanoCoin={cardanoCoin}
+                asset={asset as AssetInput}
```
Comment (mirceahasegan) [inline packages/nami/src/ui/app/components/assetsViewer.tsx:140]: Why cast? `Asset` component expects `NamiAsset`, not `AssetInput`

File: packages/nami/src/ui/app/components/asset.tsx (hunk 10 above)
Comment (mchappell) [inline packages/nami/src/ui/app/components/asset.tsx:191]: should this be `NamiAsset` ?

File: packages/nami/src/ui/app/pages/settings.tsx [hunk 12]
```diff
@@ -114,7 +117,9 @@ const Settings = ({
           aria-label="back button"
           rounded="md"
           onClick={() => {
-            history.goBack();
+            history.push(
+              location.pathname === '/settings/' ? '/' : '/settings/',
```
Comment (mchappell) [inline packages/nami/src/ui/app/pages/settings.tsx:121]: `-1` also handles previous from history

File: packages/nami/src/ui/app/components/termsOfUse.tsx [hunk 13]
```diff
@@ -636,7 +649,9 @@ const TermsOfUse = React.forwardRef((props, ref) => {
                       color="teal"
                       isExternal
                       textDecoration="underline"
-                      onClick={() => openExternalLink('https://iohk.io/en/contact/')}
+                      onClick={() => {
+                        openExternalLink('https://iohk.io/en/contact/');
```
Comment (mchappell) [inline packages/nami/src/ui/app/components/termsOfUse.tsx:653]: might be worth extracting to `.env` at some point for all of these static URLS

File: packages/nami/src/ui/app/pages/settings.tsx (hunk 12 above)
Comment (vetalcore) [reply packages/nami/src/ui/app/pages/settings.tsx:121]: it gets problematic when you reopen nami and the history is clear so you cannot actually go back

File: packages/nami/src/ui/app/components/asset.tsx (hunk 10 above)
Comment (vetalcore) [reply packages/nami/src/ui/app/components/asset.tsx]: done here 3dc6fdda24fc723b9739a11ecd6bf23b3b9ef2fe

File: packages/nami/src/ui/app/components/assetsViewer.tsx (hunk 11 above)
Comment (vetalcore) [reply packages/nami/src/ui/app/components/assetsViewer.tsx]: done here 3dc6fdda24fc723b9739a11ecd6bf23b3b9ef2fe

File: packages/nami/src/features/ada-handle/config.ts [hunk 14]
```diff
@@ -3,7 +3,10 @@ import { Wallet } from '@lace/cardano';
 
 export const ADA_HANDLE_POLICY_ID = Wallet.ADA_HANDLE_POLICY_ID;
 export const isAdaHandleEnabled = process.env.USE_ADA_HANDLE === 'true';
-export const HANDLE_SERVER_URLS: Record<Cardano.NetworkMagics, string> = {
+export const HANDLE_SERVER_URLS: Record<
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply packages/nami/src/features/ada-handle/config.ts]: done here 3dc6fdda24fc723b9739a11ecd6bf23b3b9ef2fe

File: packages/nami/src/features/ada-handle/useHandleResolver.ts [hunk 15]
```diff
@@ -10,7 +11,7 @@ export const useHandleResolver = (
   networkMagic: Cardano.NetworkMagics,
 ): HandleProvider => {
   return useMemo(() => {
-    const serverUrl = HANDLE_SERVER_URLS[networkMagic];
+    const serverUrl = HANDLE_SERVER_URLS[networkMagic]!;
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply packages/nami/src/features/ada-handle/useHandleResolver.ts]: done here 3dc6fdda24fc723b9739a11ecd6bf23b3b9ef2fe

File: packages/nami/src/ui/app/components/termsOfUse.tsx (hunk 13 above)
Comment (vetalcore) [reply packages/nami/src/ui/app/components/termsOfUse.tsx:653]: @mchappell do you mind if we do it in the next iteration? (once we need to update these urls)?

File: packages/nami/src/adapters/transactions.ts [hunk 16]
```diff
@@ -299,7 +299,7 @@ export const useTxInfo = (
   const currentAddress = walletAddresses[0];
 
   useEffect(() => {
-    if (!protocolParameters) return;
+    if (!protocolParameters || 'cbor' in tx) return;
```
Comment (vetalcore) [inline packages/nami/src/adapters/transactions.ts:302]: @mirceahasegan @mchappell `cbor' in tx` not the best way to check for `inFlight` tx?

Comment (mirceahasegan) [reply packages/nami/src/adapters/transactions.ts:302]: I think you could go the other way around and check if it's a HydratedTx. It should have one of these:
```
    index: number;
    blockHeader: PartialBlockHeader;
```

Comment (vetalcore) [reply packages/nami/src/adapters/transactions.ts]: done here 0b4f057bbc70f8623c76b11d4b9867c3c2e736ac

**Review by mirceahasegan — APPROVED** [review]
lgtm

**Pull request merged**